src/client/datascience/jupyter/liveshare/hostJupyterServer.ts

```diff
@@ -317,6 +324,8 @@ export class HostJupyterServer extends LiveShareParticipantHost(JupyterServerBas
         // Do this only if we don't have any kernel connection information, or the resource's interpreter is different.
         let changedKernel = false;
         if (
+            // For local connections this code path is not executed for native notebooks (hence only for remote).
+            (isResourceNativeNotebook(resource, this.vscodeNotebook) && !launchInfo.connectionInfo.localLaunch) ||
```

Comment:
Why is this not used for native notebooks? Seems weird. Is there more you could add to the comment? Or maybe something isn't workign the way we'd really want? (Maybe this is a side effect of the native notebook stuff halfway using the jupyterNotebook class.)

Reply:
Replied to this comment 2 times already, choppy network connection.
When using native notebooks with location connections, for some reason this path is not followed, no idea. didn't bother trying to identify, after all, this is a higher level class & didn't care about it as we're not too concerned about live share for native notebooks (not yet & possible we won't need these).

Reply:
> Maybe this is a side effect of the native notebook stuff halfway using the jupyterNotebook class.

Yes.